server/queues/tasks/ExportMarkdownZipTask.ts

```diff
@@ -70,20 +87,24 @@ export default class ExportMarkdownZipTask extends BaseTask<Props> {
         url,
       });
 
-      await ExportSuccessEmail.schedule({
-        to: user.email,
-        id: fileOperation.id,
-        teamUrl: team.url,
-      });
+      if (sendEmail) {
```

Comment:
Could we move this unto a `beforeSend` block on the emails themselves, see here as an example:

https://github.com/outline/outline/blob/863f22750ffb86fec628901a8380d285e5c12ab3/server/emails/templates/InviteAcceptedEmail.tsx#L27

Trying to go this way in the future as it feels more maintainable (the logic follows the email around irregardless of if it's triggered from multiple spots in the future.

Reply:
Sounds good.

Is is okay to omit `teamId` from the query, or is this neccessary?

```ts
where: {
    userId: id,
    teamId: fileOperation.teamId, // omitting this line
    event: "emails.export_completed",
},

Reply:
Better to keep it, as the index is `team_id_user_id`, which makes me think that code above that I referenced really needs changing. You'll have to pass `teamId` to the email.schedule call but that's okay.